common/core/web/keyboard-processor/src/text/keyboardProcessor.ts

```diff
@@ -367,6 +364,22 @@ namespace com.keyman.text {
       return modifier;
     }
 
+    /**
+     * Get state key state from layer id
```

Comment:
Is the double `state` intentional?

Reply:
Yes, on 101 key keyboards, the 'state' keys are the caps lock, num lock, scroll lock keys, but we only care about Caps Lock. So this is retrieving the state of these keys 😆.

Both `getStateFromLayer` and `getModifierState` above could be better; right now a layer called "foolscapsymbols" (okay, it's a stretch) would also be considered a 'caps lock' layer.